src/main/java/com/volmit/iris/core/TreeManager.java

```diff
@@ -0,0 +1,308 @@
+        saplingPlane.forEach(block -> block.setType(Material.AIR));
+
+        // Get object from placer
+        IrisObject object = worldAccess.getData().getObjectLoader().load(placement.getPlace().getRandom(RNG.r));
+
+        // Create object placer
+        IObjectPlacer placer = new IObjectPlacer() {
+
+            @Override
+            public int getHighest(int x, int z) {
+                return event.getWorld().getHighestBlockYAt(x, z);
+            }
+
+            @Override
+            public int getHighest(int x, int z, boolean ignoreFluid) {
+                return event.getWorld().getHighestBlockYAt(x, z, ignoreFluid ? HeightMap.OCEAN_FLOOR : HeightMap.WORLD_SURFACE);
+            }
+
+            @Override
+            public void set(int x, int y, int z, BlockData d) {
+                Block b = event.getWorld().getBlockAt(x, y, z);
+
+                // Set the block
+                b.setBlockData(d, false);
+
+                // Tell bukkit what you're doing here
+                event.getBlocks().add(b.getState());
+            }
+
+            @Override
+            public BlockData get(int x, int y, int z) {
+                return event.getWorld().getBlockAt(x, y, z).getBlockData();
+            }
+
+            @Override
+            public boolean isPreventingDecay() {
+                return true;
+            }
+
+            @Override
+            public boolean isSolid(int x, int y, int z) {
+                return get(x,y,z).getMaterial().isSolid();
+            }
+
+            @Override
+            public boolean isUnderwater(int x, int z) {
+                return false;
+            }
+
+            @Override
+            public int getFluidHeight() {
+                Engine engine;
+                if (worldAccess.getCompound().getSize() > 1) {
+                    engine = worldAccess.getCompound().getEngine(0);
+                } else {
+                    engine = (Engine) worldAccess.getCompound().getRootDimension();
+                }
+                return engine.getDimension().getFluidHeight();
+            }
+
+            @Override
+            public boolean isDebugSmartBore() {
+                return false;
+            }
+
+            @Override
+            public void setTile(int xx, int yy, int zz, TileData<? extends TileState> tile) {
+
+            }
+        };
+
+        // TODO: Prevent placing on claimed blocks (however that's defined, idk)
+
+        // TODO: Prevent placing object when overriding blocks
+
+        // Cancel the vanilla placement
+        event.setCancelled(true);
+
+        // Send out a new event
+        List<BlockState> blockStateList = new KList<>();
+        saplingPlane.forEach(b -> blockStateList.add(b.getState()));
+        StructureGrowEvent iGrow = new StructureGrowEvent(event.getLocation(), event.getSpecies(), event.isFromBonemeal(), event.getPlayer(), blockStateList);
+        Bukkit.getServer().getPluginManager().callEvent(iGrow);
+
```

Comment:
After you have called the event, check if you should update all the states you cached (the ones in the event you fired)

```java
if(!iGrow.isCancelled())
{
    for(BlockState i : iGrow.getStateList())
    {
        i.update(true, false);
    }
}
```

Reply:
Added to the sync task after calling the grow event